### ISSUE TITLE
Help entries for some builtins

### ISSUE THREAD

**Issue by joehuchette** [JuliaLang/julia#12438]
When queried with  ``help?> foo``, the following builtins lead to syntax errors:
* [x] abstract  #13177
* [x] baremodule  #13177
* [x] bitstype  #13177
* [x] for
* [ ] global   #13217
* [x] if
* [ ] importall   #13217
* [ ] let   #13217
* [ ] local   #13217
* [x] macro  #13177
* [x] module  #13177
* [ ] quote  #13217
* [x] try
* [x] typealias
* [x] while
* [ ] `'` (ctranspose); interestingly `.'` is a valid identifier, but not used in the `.'` syntax. It also needs documentation.   #13217
* [x] catch
* [x] else
* [x] elseif
* [x] end
* [x] finally

**Comment by MichaelHatherly** [comment]
Keywords are being documented at the top of [base/docs/basedocs.jl](https://github.com/JuliaLang/julia/blob/master/base/docs/basedocs.jl). Until we've covered all of them a nicer error message for this would be great to have for this.

**Comment by sbromberger** [comment]
Semi-related but not included above:

```
help?> ?
search:

  Welcome to Julia 0.4.0-dev+6817. The full manual is available at

  http://docs.julialang.org/

  as well many great tutorials and learning resources:

  http://julialang.org/learning/

  For help on a specific function or macro, type ? followed by its name, e.g. ?fft, ?@time or ?html, and press enter.

# ok, let's try the last one:

help?> html
search: HTML @html_str gethostname

Couldn't find html
Perhaps you meant htol, hton, hcat, hist, hash, hex, chol, chomp, ltoh, ntoh, call, eval, @cmd, @time, Cmd, Dims, Val, all or atan
```

**Comment by mbauman** [comment]
@joehuchette I hope you don't mind that I edited your list to add `'` and make it a checklist.  I can revert it back if you'd prefer.

**Comment by yuyichao** [comment]
FWIW, `stagedfunction` also give a syntax error.... Probably not worth documenting but no error would be nice...

**Comment by hayd** [comment]
`@generated` could do with a docstring. I'm not sure where/whether deprecated syntax help should be added (maybe it could just say "Deprecated. See `@generated`."), but IMO it's fine without...

**Comment by jakebolewski** [comment]
closed by 9d8571f57e496350d2229a4517cc270439c0067f

**Comment by tkelman** [comment]
backported in db299834118b0bc8d34a934398e6ff25d5c71916
